.github/wiki/Testing.md

```diff
@@ -0,0 +1,62 @@
+NimblePhxGenTemplate is using Github Action, the workflow files are located under `.github/workflows/` folder, it's including the Template test and Variant test workflow.
+
+#### 1/ Template test
+
+All files are located under `test/` folder.
+
+```
+.
+├── ...
+├── test
+│   ├── ...
+│   ├── nimble.phx.gen.template
+│   │   └── addons
+│   │   │   ├── ...
+│   │   │   ├── common_addon_test.exs
+│   │   │   └── variants
+│   │   │   │   └── api
+│   │   │   │   │   ├── ...
+│   │   │   │   │   └── api_addon_test.exs
+│   │   │   │   └── web
+│   │   │   │   │   ├── ...
+│   │   │   │   │   └── web_addon_test.exs
+```
+
+#### 2/ Variant test
+
+##### 2.1/ Variant
+
+NimblePhxGenTemplate is supporting 2 variants:
+
+- API
+- Web
+
+##### 2.2/ Phoenix project
+
+The Phoenix project could be either a Web project or API project.
+
+- Web project is including HTML and Webpack configuration, which was generated by `mix phx.new AppName`.
+
+- API project is NOT including HTML and Webpack configuration, which was generated by `mix phx.new AppName --no-html --no-webpack`.
+
+Aside from that, it could also be a Custom project, which contains the custom OTP app name or custom module name.
+
+- `mix phx.new AppName --module=CustomModuleName`
+- `mix phx.new AppName --app=custom_otp_app_name`
+- `mix phx.new AppName --module=CustomModuleName --app=custom_otp_app_name`
+
+So it ends up with 4 project types:
+
+- Standard Web project (`mix phx.new AppName`)
+- Custom Web project (`mix phx.new AppName --module=CustomModuleName --app=custom_otp_app_name`)
+- Standard API project (`mix phx.new AppName --no-html --no-webpack`)
+- Custom API project (`mix phx.new AppName --no-html --no-webpack --module=CustomModuleName --app=custom_otp_app_name`)
+
+Putting it all together, it is 6 variant test cases.
+
+- Applying the `API variant` to a `Standard Web project`
+- Applying the `API variant` to a `Custom Web project`
+- Applying the `API variant` to a `Standard API project`
+- Applying the `API variant` to a `Custom API project`
+- Applying the `Web variant` to a `Standard Web project`
+- Applying the `Web variant` to a `Custom Web project`
```

Comment:
I remember on the main README file we have LiveView, could you check the latest README file @byhbt 

```
- Applying the `Live variant` to a `Standard LiveView project`
- Applying the `Live variant` to a `Custom LiveView project`
```

Reply:
ok, i synced with the latest content from the Repo Readme: https://github.com/nimblehq/elixir-templates/pull/135/commits/f9ee816b8de9c4fa7106d45944d69bf25e2ba92b

I have a thought that should we remove the test content from the Repo README.md? 🤔  because we already have it in the wiki. What do you think?

Reply:
We can remove the content from README @byhbt, just need to put a link to WIKI on README file so the people can see that.

eg: 

```
## Project documentation

Most of the documentation locate in the `.github/wiki` directory, which is published to the [project's wiki](https://github.com/xxxxx/wiki).
```

Reply:
yeah, that's also ok, for now i just put https://github.com/nimblehq/elixir-templates/pull/135/commits/116858599297b87ab49d5c5f55c4788c68237759

Maybe we need to refine the README again after we have a Wiki section.

Reply:
I have added a comment https://github.com/nimblehq/elixir-templates/pull/135#discussion_r774311290